lib/matplotlib/axes/_base.py

```diff
@@ -300,29 +335,10 @@ def __call__(self, axes, *args, data=None, return_kwargs=False, **kwargs):
 
     def get_next_color(self):
         """Return the next color in the cycle."""
-        entry = self._cycler_items[self._idx]
-        if "color" in entry:
-            self._idx = (self._idx + 1) % len(self._cycler_items)  # Advance cycler.
-            return entry["color"]
-        else:
-            return "k"
-
-    def _getdefaults(self, kw, ignore=frozenset()):
-        """
-        If some keys in the property cycle (excluding those in the set
-        *ignore*) are absent or set to None in the dict *kw*, return a copy
-        of the next entry in the property cycle, excluding keys in *ignore*.
-        Otherwise, don't advance the property cycle, and return an empty dict.
-        """
-        defaults = self._cycler_items[self._idx]
-        if any(kw.get(k, None) is None for k in {*defaults} - ignore):
-            self._idx = (self._idx + 1) % len(self._cycler_items)  # Advance cycler.
-            # Return a new dict to avoid exposing _cycler_items entries to mutation.
-            return {k: v for k, v in defaults.items() if k not in ignore}
-        else:
-            return {}
+        return self._prop_cycle.get_next_color()
 
-    def _setdefaults(self, defaults, kw):
+    @staticmethod
```

Comment:
This is not strictly related, but noticed that `_set_defaults` does not depend on the class, so making that explicit via `staticmethod` as an improvement on the side.